code/modules/cm_marines/altitude_control_console.dm

```diff
@@ -44,34 +55,50 @@ GLOBAL_VAR_INIT(ship_alt, SHIP_ALT_MED)
 
 /obj/structure/machinery/computer/altitude_control_console/process()
 	. = ..()
+	//Updating temperature
 	var/temperature_change
-	if(GLOB.ship_temp >= OVERHEAT)
-		ai_silent_announcement("Attention: orbital correction no longer sustainable, moving to geo-synchronous orbit until engine cooloff.", ";", TRUE)
+	switch(GLOB.ship_alt)
+		if(SHIP_ALT_LOW)
+			if(prob(50))
+				temperature_change = HEATING
+			else
+				temperature_change = B_HEATING
+		if(SHIP_ALT_MED)
+			temperature_change = COOLING
+		if(SHIP_ALT_HIGH)
+			if(prob(75))
+				temperature_change = OVERHEAT_COOLING
+			else
+				temperature_change = COOLING
+	GLOB.ship_temp = clamp(GLOB.ship_temp += temperature_change, 0, OVERHEAT)
+
+	//Override orbit, announce low orbit
+	if(GLOB.ship_alt == SHIP_ALT_HIGH && GLOB.ship_temp == 0)
+		ai_silent_announcement("Attention: Engine cooloff completed, automatic stabilization to most optimal geo-synchronous orbit undergoing.", ";", TRUE)
+		GLOB.ship_alt = SHIP_ALT_MED
+		for(var/mob/living/carbon/current_mob in GLOB.living_mob_list)
+			if(!is_mainship_level(current_mob.z))
+				continue
+			current_mob.apply_effect(3, SLOW)
+			shake_camera(current_mob, 10, 2)
+		return
+	if(GLOB.ship_alt == SHIP_ALT_LOW && GLOB.ship_temp >= OVERHEAT)
+		ai_silent_announcement("Attention: Low altitude orbital maneuver no longer sustainable, moving to furthest geo-synchronous orbit until engine cooloff.", ";", TRUE)
 		GLOB.ship_alt = SHIP_ALT_HIGH
```

Comment:
No cooldown is currently applied when the ship automatically makes a change. It needs to.